plugins/lsp/language-servers/rust-analyzer-config.nix

```
@@ -1,167 +1,65 @@
+    else if kind == "number" then
+      let
+        inherit (typeInfo) minimum maximum;
+      in
       if minimum != null && maximum != null then
-        types.numbers.between minimum maximum
+        lib.types.numbers.between minimum maximum
       else if minimum != null then
-        types.addCheck types.number (x: x >= minimum)
+        lib.types.addCheck lib.types.number (x: x >= minimum)
       else if maximum != null then
-        types.addCheck types.number (x: x <= maximum)
+        lib.types.addCheck lib.types.number (x: x <= maximum)
       else
-        types.number
-    else if type == "integer" then
+        lib.types.number
+    else if kind == "integer" then
+      let
+        inherit (typeInfo) minimum maximum;
+      in
       if minimum != null && maximum != null then
-        types.ints.between minimum maximum
+        lib.types.ints.between minimum maximum
       else if minimum != null then
-        types.addCheck types.int (x: x >= minimum)
+        lib.types.addCheck lib.types.int (x: x >= minimum)
       else if maximum != null then
-        types.addCheck types.int (x: x <= maximum)
+        lib.types.addCheck lib.types.int (x: x <= maximum)
       else
-        types.int
+        lib.types.int
+    else if kind == "object" then
+      lib.types.attrsOf lib.types.anything
+    else if kind == "string" then
+      lib.types.str
+    else if kind == "boolean" then
+      lib.types.bool
     else
-      throw "unhandled type `${builtins.toJSON type}` (in ${opt})";
+      throw "Unknown type: ${kind}";
 
-  mkRustAnalyzerOption =
-    opt:
-    {
-      default,
-      markdownDescription,
-      type ? null,
-      anyOf ? null,
-      # Enum types
-      enum ? null,
-      enumDescriptions ? null,
-      # Int types
-      minimum ? null,
-      maximum ? null,
-      # List types
-      items ? null,
-      uniqueItems ? false,
-    }:
-    mkOption {
-      type = types.nullOr (
-        mkRustAnalyzerType opt {
-          inherit
-            type
-            enum
-            minimum
-            maximum
-            items
-            anyOf
-            uniqueItems
-            ;
-        }
-      );
+  mkNixOption =
+    { description, type }:
+    lib.mkOption {
       default = null;
-      description =
-        if enum != null then
-          let
-            valueDesc = builtins.map ({ fst, snd }: ''- ${fst}: ${snd}'') (
-              lists.zipLists enum enumDescriptions
-            );
-          in
-          ''
-            ${markdownDescription}
-
-            Values:
-            ${builtins.concatStringsSep "\n" valueDesc}
-
-            ```nix
-            ${generators.toPretty { } default}
-            ```
-          ''
-        else
-          ''
-            ${markdownDescription}
-
-            default:
-            ```nix
-            ${generators.toPretty { } default}
-            ```
-          '';
+      inherit description;
+      type = lib.types.nullOr (mkRustAnalyzerType type);
```

Handle plugin default outside of the description:

```suggestion
      type = lib.types.nullOr (mkRustAnalyzerType type);
      defaultText = helpers.pluginDefaultText {
        inherit pluginDefault;
        default = null;
      };
```

---

I know we didn't do this before, but this PR is a good opportunity to fix that.

---

If you used `helpers.defaultNullOpts.mkNullable'` instead of `lib.mkOption`, then you could pass in `pluginDefault` directly, without needing to use `helpers.pluginDefaultText`.